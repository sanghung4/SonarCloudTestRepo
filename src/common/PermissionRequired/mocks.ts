import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
