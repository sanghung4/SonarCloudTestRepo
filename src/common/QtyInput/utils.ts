import { QtyAdjustmentModal } from 'common/QtyInput';

export type QtyUpdateProps = {
  allowZero?: boolean | string;
  increment: number;
  inputQty: number;
  max?: number;
  productName?: string;
  setModal: (value: QtyAdjustmentModal) => void;
};

export function qtyUpdateLogic({
  allowZero,
  increment,
  inputQty,
  max,
  productName,
  setModal
}: QtyUpdateProps) {
  const remainder = increment ? inputQty % increment : 0;

  // When there's remainder, it needs to be adjusted to be divisible by `increment`
  if (remainder) {
    const matchIncrement = inputQty - remainder + increment;
    const toAdjustTo = max
      ? Math.min(matchIncrement, maxQtyLogic(increment, max)!)
      : matchIncrement;
    setModal({
      open: true,
      adjustedQty: toAdjustTo,
      minQty: allowZero ? 0 : increment,
      productName: productName ?? ''
    });
    return toAdjustTo;
  }
  return capMaxQtyLogic(inputQty, max);
}

export function maxQtyLogic(increment: number, incomingMax?: number) {
  return incomingMax ? incomingMax - (incomingMax % increment) : undefined;
}

export function capMaxQtyLogic(inputQty: number, max?: number) {
  return max ? Math.min(inputQty, max) : inputQty;
}

export function canUpdate(
  newqty: number,
  oldqty: number,
  allowZero?: boolean | string
) {
  const notAtZero = !allowZero && newqty > 0;
  const isPositive = allowZero && newqty >= 0;
  const notAtLowest = notAtZero || isPositive;
  return Boolean(newqty !== oldqty && notAtLowest);
}
