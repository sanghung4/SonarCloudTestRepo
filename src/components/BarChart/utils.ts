export const barOptions = {
  responsive: true,
  plugins: {
    legend: {
      position: "top" as const,
    },
  },
};

export const defaultBarData = {
  labels: [],
  datasets: [
    {
      label: "Dataset 1",
      data: [],
    },
    {
      label: "Dataset 2",
      data: [],
    },
  ],
};
