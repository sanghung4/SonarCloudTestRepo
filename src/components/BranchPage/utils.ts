import { GetLocationsQuery } from 'api';
import { OldColors, Colors } from 'constants/style';

export const ButtonValues = [
  {
    title: 'Not Started',
    value: 'Not Started',
    color: Colors.SECONDARY_2100,
  },
  {
    title: 'Started',
    value: 'Started',
    color: OldColors.BLUE,
  },
  {
    title: 'Completed',
    value: 'Completed',
    color: Colors.SUPPORT_1100,
  },
];

export const getLocationData = (data?: GetLocationsQuery) => ({
  locations: data?.locations.content || [],
  totalLocations: data?.locations.totalLocations || 1,
  totalCounted: data?.locations.totalCounted || 0,
});
