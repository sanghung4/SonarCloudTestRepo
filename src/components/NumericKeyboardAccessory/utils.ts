import { Appearance } from 'react-native';

export const isDark = (): boolean => {
  return Appearance.getColorScheme() === 'dark';
};

export const getValue = (index: number) => (index === 9 ? '0' : `${index + 1}`);
