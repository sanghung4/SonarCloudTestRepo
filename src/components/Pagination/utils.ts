export const DOTS = -1;

export const arrayOfRange = (start: number, end: number) => {
  const length = end - start + 1;
    /*
        Create an array of certain length and set the elements within it from
      start value to end value.
    */
    return Array.from({ length }, (_, idx) => idx + start);
  };