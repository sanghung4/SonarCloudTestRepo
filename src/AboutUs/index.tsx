import React from 'react';
import { useContext } from 'react';

import {
  Box,
  Button,
  Container,
  Grid,
  Image,
  Link,
  Typography
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { useQuery } from '@apollo/client';

import {
  BrandTitle,
  ContactUsButton,
  CenterContainer,
  DivisionCards,
  InfoSectionsCollection,
  MainBackground,
  MissionTitle,
  MissionText,
  NeedAssistanceImage,
  NeedAssistanceImageGrid,
  NeedAssistanceParagraph,
  NeedAssistanceTitle,
  Paragraph
} from 'AboutUs/util/styles';
import { Title } from 'AboutUs/util/styles';
import { aboutUsQuery } from 'AboutUs/util/query';
import { AuthContext } from 'AuthProvider';
import { useDomainInfo } from 'hooks/useDomainInfo';
import useDocumentTitle from 'hooks/useDocumentTitle';
import notfound from 'images/notfound.png';
import { generateCompanyUrl } from 'utils/brandList';
import { Configuration } from 'utils/configuration';
import Loader from 'common/Loader';
import { useState } from 'react';

const AboutUs = () => {
  /**
   * Custom Hooks
   */
  const { t } = useTranslation();
  const { port } = useDomainInfo();
  useDocumentTitle(t('common.about'));

  /**
   * Context
   */
  const { activeFeatures } = useContext(AuthContext);

  /**
   * State
   */
  const [showText, setShowText] = useState('');

  const { data, error, loading } = useQuery(aboutUsQuery, {
    context: {
      clientName: 'contentful'
    },
    variables: {
      aboutUsId: Configuration.contentfulAboutUsId ?? '',
      isPreview: Configuration.contentfulPreviewEnable
    }
  });

  const landerBackground = data?.aboutUs?.heroImage.url
    ? `linear-gradient(to right,
  rgba(0, 48, 87, 1) 10%, 
  rgba(0, 48, 87, 0) 100%
   ), url(${data?.aboutUs?.heroImage.url})`
    : undefined;

  /**
   * Callbacks
   */
  const handleMouseEnter = (i: any) => {
    setShowText(i);
  };

  /**
   * Render
   */
  if (loading) return <Loader />;
  if (error || !data?.aboutUs)
    return (
      <Box
        px={4}
        pt={10}
        pb={10}
        display="flex"
        flexDirection="column"
        alignItems="center"
        data-testid="error-message"
      >
        <Typography variant="body1" align="center">
          {t('aboutUs.error')}
        </Typography>
      </Box>
    );

  return (
    <Box
      display="flex"
      flexDirection="column"
      width="100%"
      bgcolor="common.white"
    >
      <MainBackground
        position="relative"
        style={{ backgroundImage: landerBackground }}
        data-testid="main-background"
      >
        <Container>
          <Box>{data.aboutUs.title}</Box>
          <Box pb={5} pr={16}>
            {data.aboutUs.title2}
          </Box>
        </Container>
      </MainBackground>
      <Container>
        <CenterContainer>
          <Title textAlign="left">{data.aboutUs.divisionsTitle?.title}</Title>
          <Paragraph
            fontSize="30px"
            fontWeight={400}
            fontFamily="Helvetica Neue"
            pb="5px"
          >
            {data.aboutUs.divisionsTitle?.paragraph}
          </Paragraph>
        </CenterContainer>
        <Box>
          <Grid container spacing={3}>
            {data.aboutUs.divisionsCarouselCollection?.items.map(
              (
                item: {
                  title: string;
                  paragraph: string;
                  paragraphColor: string;
                  linkText: string;
                  logo: { url: string };
                  backgroundImage: { url: string };
                },
                index?: React.Key | null
              ) => {
                const url = item?.linkText;
                const logo = item?.logo?.url;
                return (
                  <Grid
                    item
                    container
                    xs={12}
                    md={3}
                    key={index}
                    direction="column"
                    alignItems="center"
                  >
                    <DivisionCards
                      sx={{
                        bgcolor: item?.paragraphColor ?? 'common.white',
                        pt: '50',
                        pb: '30',
                        color: 'common.white',
                        backgroundImage: `url(${item?.backgroundImage?.url})`,
                        backgroundRepeat: 'no-repeat',
                        backgroundSize: 'cover',
                        boxShadow: '4',
                        '&:hover': {
                          backgroundImage: 'none'
                        }
                      }}
                      pt={50}
                      pb={30}
                      color="common.white"
                      boxShadow={4}
                      onMouseEnter={() => handleMouseEnter(index)}
                      onMouseLeave={() => setShowText('')}
                    >
                      {showText === index ? (
                        <Grid>
                          <Typography
                            display="block"
                            pl="30"
                            fontWeight={500}
                            fontSize="35px"
                            lineHeight="35px"
                            fontFamily="Helvetica Neue"
                            textAlign="left"
                          >
                            {item?.title}
                          </Typography>
                          <Typography
                            display="block"
                            fontWeight={400}
                            fontSize="18px"
                            lineHeight="24.5px"
                            fontFamily="Helvetica Neue"
                            pl="2px"
                            pt={2}
                            textAlign="left"
                          >
                            {item?.paragraph}
                          </Typography>

                          <Typography
                            textAlign="left"
                            pt="20px"
                            pb="20px"
                            sx={{ textDecoration: 'underline' }}
                          >
                            <Link color="common.white" href="/location-search">
                              {url}
                            </Link>
                          </Typography>
                        </Grid>
                      ) : (
                        <Grid container direction="column" alignItems="center">
                          <Grid item pb="25px">
                            <Typography display="initial">
                              <Image
                                src={logo}
                                alt="Logo"
                                fallback={notfound}
                              />
                            </Typography>
                          </Grid>
                          <Grid item>
                            <Typography
                              fontWeight={400}
                              fontSize="35px"
                              lineHeight="35px"
                              fontFamily="Helvetica Neue"
                              display="block"
                            >
                              {item?.title}
                            </Typography>
                          </Grid>
                        </Grid>
                      )}
                    </DivisionCards>
                  </Grid>
                );
              }
            )}
          </Grid>
        </Box>
      </Container>

      <CenterContainer>
        {data.aboutUs.infoSectionsCollection?.items.map(
          (
            item: { paragraph: string; text: string; title: string },
            i?: React.Key | null
          ) => {
            const paragraph = item.paragraph;
            const text = item.text;
            return (
              <InfoSectionsCollection key={`division-${i}`}>
                <MissionTitle
                  fontFamily={!i ? 'Helvetica Neue' : 'Permanent Marker'}
                  pl={10}
                >
                  {item.title}
                </MissionTitle>
                <Paragraph pb={2} pt={3}>
                  {paragraph}
                </Paragraph>
                <MissionText>{text}</MissionText>
              </InfoSectionsCollection>
            );
          }
        )}
      </CenterContainer>

      <Box
        textAlign="center"
        lineHeight="24px"
        whiteSpace="break-spaces"
        color="common.black"
        bgcolor="secondary04.main"
        padding="40px"
      >
        <Grid>
          <BrandTitle>{data.aboutUs.brands.title}</BrandTitle>
        </Grid>
        <Grid container>
          {data.aboutUs.brands?.listCollection?.items.map(
            (
              item: {
                url: string;
                logo: { url: string };
                statesList: [];
                urlText: string;
              },
              i?: React.Key | null
            ) => {
              const url = item.url;
              const statesList = item.statesList;
              const logo = item.logo.url;
              const urlText = item.urlText;
              return (
                <Grid
                  item
                  container
                  xs={12}
                  md={4}
                  key={`brand-${i}`}
                  direction="column"
                  alignItems="center"
                  justifyContent="center"
                  data-testid={`brand-company-${i}`}
                >
                  <Grid
                    item
                    container
                    direction="column"
                    alignItems="center"
                    justifyContent="center"
                    py={5}
                  >
                    <Link
                      href={generateCompanyUrl(
                        url,
                        port,
                        Configuration.environment,
                        !!activeFeatures?.includes('WATERWORKS')
                      )}
                      data-testid={`brand-url-${i}`}
                    >
                      <Image
                        src={logo}
                        alt="Logo"
                        data-testid={`brand-logo-${i}`}
                        fallback={notfound}
                      />
                    </Link>
                    <Grid
                      alignItems="center"
                      justifyContent="center"
                      data-testid={`brand-countries-${i}`}
                      pt={2}
                    >
                      <Typography>{statesList.join(', ')}</Typography>
                    </Grid>
                    <Grid item pt={3}>
                      <Link
                        href={generateCompanyUrl(
                          url,
                          port,
                          Configuration.environment,
                          !!activeFeatures?.includes('WATERWORKS')
                        )}
                      >
                        <Button
                          variant="primary"
                          data-testid={`brand-shopnow-${i}`}
                        >
                          <Typography data-testid={`brand-url-${i}`}>
                            {urlText}
                          </Typography>
                        </Button>
                      </Link>
                    </Grid>
                  </Grid>
                </Grid>
              );
            }
          )}
        </Grid>
      </Box>
      <Grid container sx={{ backgroundColor: 'primary.main' }}>
        <Grid item xs={12} md={12} lg={7} alignItems="flex-end">
          <NeedAssistanceImageGrid>
            <Image
              src={data.aboutUs.needAssistanceBlock?.backgroundImage.url}
              alt="Background"
              sx={{ maxWidth: '100%', maxHeight: '60%' }}
              fallback={notfound}
            />
          </NeedAssistanceImageGrid>
        </Grid>
        <Grid
          item
          xs={12}
          md={5}
          alignItems="flex-end"
          color="common.white"
          textAlign="center"
        >
          <Grid container justifyContent="center">
            <NeedAssistanceImage
              fallback={notfound}
              src={data.aboutUs.needAssistanceBlock?.logo.url}
              alt="Logo"
            />
          </Grid>
          <NeedAssistanceTitle>
            {data.aboutUs.needAssistanceBlock?.title}
          </NeedAssistanceTitle>
          <NeedAssistanceParagraph>
            {data.aboutUs.needAssistanceBlock?.paragraph}
          </NeedAssistanceParagraph>
          <Grid container justifyContent="center" pb="50px">
            <Link href="/support">
              <ContactUsButton
                border={2}
                borderLeft={2}
                borderRight={2}
                marginTop="20px"
                margin={5}
              >
                {data.aboutUs.needAssistanceBlock?.linkText}
              </ContactUsButton>
            </Link>
          </Grid>
        </Grid>
      </Grid>
    </Box>
  );
};

export default AboutUs;
