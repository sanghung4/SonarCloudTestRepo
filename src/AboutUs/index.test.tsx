import { waitFor } from '@testing-library/react';
import AboutUs from 'AboutUs';
import { mockNullResponse, success } from 'AboutUs/tests/mocks';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';
import { Configuration } from 'utils/configuration';

describe('AboutUs', () => {
  it('Should match snapshop on desktop', () => {
    setBreakpoint('desktop');
    const { container } = render(<AboutUs />);
    expect(container).toMatchSnapshot();
  });
  it('Should match snapshop on mobile', () => {
    setBreakpoint('mobile');
    const { container } = render(<AboutUs />);
    expect(container).toMatchSnapshot();
  });

  it('Should render aboutus page on desktop with contentful data', async () => {
    setBreakpoint('desktop');
    Configuration.contentfulAboutUsId = '13j7MQQu4r6gSXv6KLiTYX';
    Configuration.contentfulPreviewEnable = false;
    const { getByTestId } = render(<AboutUs />, {
      mocks: [success]
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(getByTestId('main-background')).toBeInTheDocument();
  });

  it('Should render aboutus page on mobile with contentful data', async () => {
    setBreakpoint('mobile');
    Configuration.contentfulAboutUsId = '13j7MQQu4r6gSXv6KLiTYX';
    Configuration.contentfulPreviewEnable = false;
    const { getByTestId } = render(<AboutUs />, {
      mocks: [success]
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(getByTestId('main-background')).toBeInTheDocument();
  });

  it('Should render error message on desktop without contentful data', async () => {
    setBreakpoint('desktop');
    Configuration.contentfulAboutUsId = '13j7MQQu4r6gSXv6KLiTYX';
    Configuration.contentfulPreviewEnable = true;
    const { getByTestId } = render(<AboutUs />, {
      mocks: [success]
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(getByTestId('error-message')).toBeInTheDocument();
  });

  it('Should render error message on mobile with contentful data', async () => {
    setBreakpoint('mobile');
    Configuration.contentfulAboutUsId = 'A13j7MQQu4r6gSXv6KLiTYX';
    Configuration.contentfulPreviewEnable = false;
    const { getByTestId } = render(<AboutUs />, {
      mocks: [mockNullResponse]
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(getByTestId('error-message')).toBeInTheDocument();
  });
});
