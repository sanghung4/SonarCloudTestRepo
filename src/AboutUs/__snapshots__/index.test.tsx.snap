// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs Should match snapshop on desktop 1`] = `
<div>
  <div
    class="MuiBox-root css-1nen2z4"
    data-testid="loader-component"
  >
    <div
      class="MuiBox-root css-hyxlzm"
    >
      <span
        aria-valuenow="100"
        class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px; transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
            style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
          />
        </svg>
      </span>
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1gse1c1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`AboutUs Should match snapshop on mobile 1`] = `
<div>
  <div
    class="MuiBox-root css-1nen2z4"
    data-testid="loader-component"
  >
    <div
      class="MuiBox-root css-hyxlzm"
    >
      <span
        aria-valuenow="100"
        class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px; transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
            style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
          />
        </svg>
      </span>
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1gse1c1-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="19.5"
            stroke-width="5"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
